### PR TITLE
Document deduplication tools using pyspark.

### PR DESCRIPTION
**Pull request by garyzhang99**
- Added tools for text deduplication, following the description in GPT3 paper.
- Use pyspark for distributed setting.

- [x] Add Readme.md
- [x] Test on larger scale

### CODE REVIEW

File: tools/distributed_deduplication/spark_dedup.py [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+        logger.info('No components found.')
+        sys.exit(0)
+
+    components = spark.createDataFrame(results,
+                                       schema=['id', 'component'
+                                               ]).sort(['component', 'id'])
+    components.show()
+    df = df.join(components, on='id', how='left')
+    df = df.filter(F.col('component').isNull()).drop('id', 'component').cache()
+    # df.write.mode("overwrite").json("output_of_spark")
```
Comment (pan-x-c) [inline tools/distributed_deduplication/spark_dedup.py:91]: Remove this comment, others LGTM

File: tools/distributed_deduplication/spark_dedup.py [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+def dedup_dataset(dataset_path: str,
+                  result_path: str,
+                  tokenizer: Union[str, None] = None,
+                  threshold: float = 0.7,
+                  num_features: int = 1047576,
+                  num_hashtables: int = 10,
+                  text_key: str = 'text'):
+    """
+    Perform fuzzy text deduplication on the given dataset.
+    :param dataset_path: the path to the dataset you want to predict for
```
Comment (pan-x-c) [inline tools/distributed_deduplication/spark_dedup.py:27]: check the description here

File: tools/distributed_deduplication/spark_dedup.py (hunk 1 above)
Comment (garyzhang99) [reply tools/distributed_deduplication/spark_dedup.py]: Done

File: tools/distributed_deduplication/spark_dedup.py (hunk 2 above)
Comment (garyzhang99) [reply tools/distributed_deduplication/spark_dedup.py]: Modified

File: tools/distributed_deduplication/README.md [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+```shell
+python spark_dedup.py \
+    <dataset_path> \
+    <result_path> \
+    [--tokenizer <tokenizer_type>] \
+    [--threshold <threshold_value>] \
+    [--num_features <num_features>] \
+    [--num_hashtables <num_hashtables>] \
+    [--text_key <text_key>] \
+    [--overall_stats]
```
Comment (HYLcool) [inline tools/distributed_deduplication/README.md:33]: overall_stats这个参数在这个工具里应该没有了

File: tools/distributed_deduplication/README.md [hunk 4]
```diff
@@ -0,0 +1,46 @@
+# Distributed Fuzzy Deduplication Tools
```
Comment (HYLcool) [inline tools/distributed_deduplication/README.md:1]: 可以添加一个对应的中文README

File: tools/distributed_deduplication/README.md (hunk 3 above)
Comment (garyzhang99) [reply tools/distributed_deduplication/README.md]: Removed

File: tools/distributed_deduplication/README.md (hunk 4 above)
Comment (garyzhang99) [reply tools/distributed_deduplication/README.md:1]: Added

File: tools/distributed_deduplication/README_ZH.md [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+    <result_path> \
+    [--tokenizer <tokenizer_type>] \
+    [--threshold <threshold_value>] \
+    [--num_features <num_features>] \
+    [--num_hashtables <num_hashtables>] \
+    [--text_key <text_key>] \
+    [--master_url <master_url>]
+# 打印使用信息
+python spark_dedup.py --help
+
```
Comment (HYLcool) [inline tools/distributed_deduplication/README_ZH.md:27]: 这儿可能缺了个 ```

File: tools/distributed_deduplication/spark_dedup.py [hunk 6]
```diff
@@ -21,7 +22,8 @@ def dedup_dataset(dataset_path: str,
                   threshold: float = 0.7,
                   num_features: int = 1047576,
                   num_hashtables: int = 10,
-                  text_key: str = 'text'):
+                  text_key: str = 'text',
+                  master_url: str = 'text'):
```
Comment (HYLcool) [inline tools/distributed_deduplication/spark_dedup.py:26]: master_url默认值要不置为None？

Comment (garyzhang99) [reply tools/distributed_deduplication/spark_dedup.py]: Sorry, typo

**Review by HYLcool — APPROVED** [review]
LGTM 👍🏻

**Pull request merged**